parsons/scytl/scytl.py

```diff
@@ -0,0 +1,664 @@
+
+                    if choice.tag == "ParticipatingCounties":
+                        continue
+
+                    source_cand_data = choice.attrib
+                    cand_name = source_cand_data.get('text')
+                    cand_party = source_cand_data.get('party')
+
+                    for vote_type in choice:
+                        vote_type_label = vote_type.attrib['name']
+
+                        for county in vote_type:
+                            county_name = county.attrib["name"]
+                            cand_votes[county_name] = int(county.attrib["votes"])
+
+                            county_turnout = county_dict.get(county_name) or {}
+
+                            result = {
+                                'state': state,
+                                'county_name': county_name,
+                                'office': office,
+                                'ballots_cast': county_turnout.get('ballotsCast'),
+                                'reg_voters': county_turnout.get('totalVoters'),
+                                'precincts_reporting': county_turnout.get('precinctsReported'),
+                                'total_precincts': county_turnout.get('precinctsParticipating'),
+                                'vote_method': vote_type_label,
+                                'candidate_name': cand_name,
+                                'candidate_party': cand_party,
+                                'recorded_votes': cand_votes[county_name],
+                                'timestamp_last_updated': timestamp
+                            }
+
+                            county_votes.append(result)
+
+        return county_votes
+
+    def _fetch_and_parse_summary_results(
+        self, administrator: str, election_id: str, version_num: str, county=''
+    ) -> list[object]:
+        """
+        Fetches the summary results CSV file from the Scytl site and parses it
+        into a list of election results by candidate.
+
+        `Args`:
+            administrator: str
+                The url code for the election administrator, either the two-letter
+                state code or the state code and the county, separated by a slash
+            election_id: str
+                The election id for the given election as a string
+            version_num: str
+                The latest version ID of the election as a string
+            county: str
+                The name of the county associated with the summary file
+        `Returns`:
+            list[object]
+            The list of election results by candidate.
+        """
+
+        summary_csv_zip_url = summary_csv_url_template.format(
+            administrator=administrator, election_id=election_id, version_num=version_num
+        )
+
+        zip_bytes = self._parse_file_from_zip_url(summary_csv_zip_url, 'summary.csv')
+
+        if not zip_bytes:
+            return []
+
+        string_buffer = StringIO(zip_bytes.decode('latin-1'))
```

Comment:
recommended: use context managers to ensure that these get cleaned up on function return.